src/Pos/Ssc/GodTossing/Toss/Pure.hs

```diff
@@ -63,6 +67,21 @@ instance MonadToss PureToss where
     resetShares = PureToss $ gsShares .= mempty
     setEpochOrSlot eos = PureToss $ gsVssCertificates %= VCD.setLastKnownEoS eos
 
+runPureToss
+    :: LoggerName
+    -> MultiRichmenStake
+    -> GtGlobalState
+    -> PureToss a
+    -> (a, GtGlobalState, [LogEvent])
+runPureToss loggerName richmenData gs =
+    convertRes .
+    (\a -> runRWS a richmenData gs) .
+    usingLoggerName loggerName . runPureLog . runNamedPureLogger . getPureToss
```

Comment:
There're some utility functions that helps reduce code amount:
https://github.com/serokell/log-warper/blob/c2a21fd2971d9ed48cdd7295f8efd56fc8ec4821/src/System/Wlog/CanLog.hs#L155